certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -240,23 +233,30 @@ def choose_vhost(self, target_name, raise_if_no_match=True):
         matches = self._get_ranked_matches(target_name)
         vhost = self._select_best_name_match(matches)
         if not vhost:
-            if raise_if_no_match:
+            if create_if_no_match:
+                vhost = self._vhost_from_duplicated_default(target_name)
+            else:
                 # No matches. Raise a misconfiguration error.
                 raise errors.MisconfigurationError(
                             ("Cannot find a VirtualHost matching domain %s. "
                              "In order for Certbot to correctly perform the challenge "
                              "please add a corresponding server_name directive to your "
                              "nginx configuration: "
                              "https://nginx.org/en/docs/http/server_names.html") % (target_name))
-            else:
-                return None
-        else:
-            # Note: if we are enhancing with ocsp, vhost should already be ssl.
-            if not vhost.ssl:
-                self._make_server_ssl(vhost)
+        # Note: if we are enhancing with ocsp, vhost should already be ssl.
+        if not vhost.ssl:
+            vhost = self._duplicate_and_make_server_ssl(vhost)
```

Comment:
I think it's cleaner, more consistent, and easier to understand this way.

Plus, if they later want to manually remove or redirect their 80 site, it's easier to make those manual changes and more clear how to do so.

Reply:
I think most of that is largely subjective. I buy setting up the redirect is easier, but we automate that for users anyway. To stop listening on port 80 all together, I'm not sure deleting a server block is easier/clearer than deleting `listen 80;`.

Have any users expressed confusion over this or asked we change this behavior? I couldn't find any. If the reason you want this is coming from an argument that it's better for the user, want to ask other Certbot devs or tech @ EFF? If we do this, we should agree on the wording beforehand to avoid any bias.